casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala

```diff
@@ -106,38 +110,43 @@ object BlockApproverProtocol {
+        validators = blockBonds.toSeq.map(b => ProofOfStakeValidator(b._1.toByteArray, b._2))
+        posParams  = ProofOfStakeParams(minimumBond, maximumBond, validators)
+        faucetCode = if (faucet) Faucet.basicWalletFaucet(_) else Faucet.noopFaucet
+        genesisBlessedContracts = Genesis
+          .defaultBlessedTerms(timestamp, posParams, wallets, faucetCode)
+          .toSet
+        blockDeploys          = body.deploys.flatMap(ProcessedDeployUtil.toInternal)
+        genesisBlessedTerms   = genesisBlessedContracts.flatMap(_.term)
+        genesisBlessedDeploys = genesisBlessedContracts.flatMap(_.raw)
+        _ <- blockDeploys
+              .forall(
+                d =>
+                  genesisBlessedTerms.contains(d.deploy.term.get) && genesisBlessedDeploys
+                    .exists(dd => deployDataEq.eqv(dd, d.deploy.raw.get))
+              )
+              .either(())
+              .or("Candidate deploys do not match expected deploys.")
+        _ <- (blockDeploys.size == genesisBlessedContracts.size)
+              .either(())
+              .or("Mismatch between number of candidate deploys and expected number of deploys.")
+      } yield (blockDeploys, postState)
     for {
-      _ <- (candidate.requiredSigs == requiredSigs)
-            .either(())
-            .or("Candidate didn't have required signatures number.")
-      block      <- Either.fromOption(candidate.block, "Candidate block is empty.")
-      body       <- Either.fromOption(block.body, "Body is empty")
-      postState  <- Either.fromOption(body.state, "Post state is empty")
-      blockBonds = postState.bonds.map { case Bond(validator, stake) => validator -> stake }.toMap
-      _ <- (blockBonds == bonds)
-            .either(())
-            .or("Block bonds don't match expected.")
-      validators = blockBonds.toSeq.map(b => ProofOfStakeValidator(b._1.toByteArray, b._2))
-      posParams  = ProofOfStakeParams(minimumBond, maximumBond, validators)
-      faucetCode = if (faucet) Faucet.basicWalletFaucet(_) else Faucet.noopFaucet
-      genesisBlessedContracts = Genesis
-        .defaultBlessedTerms(timestamp, posParams, wallets, faucetCode)
-        .toSet
-      blockDeploys          = body.deploys.flatMap(ProcessedDeployUtil.toInternal)
-      genesisBlessedTerms   = genesisBlessedContracts.flatMap(_.term)
-      genesisBlessedDeploys = genesisBlessedContracts.flatMap(_.raw)
-      _ <- blockDeploys
-            .forall(
-              d =>
-                genesisBlessedTerms.contains(d.deploy.term.get) && genesisBlessedDeploys
-                  .exists(dd => deployDataEq.eqv(dd, d.deploy.raw.get))
-            )
-            .either(())
-            .or("Candidate deploys do not match expected deploys.")
-      _ <- (blockDeploys.size == genesisBlessedContracts.size)
-            .either(())
-            .or("Mismatch between number of candidate deploys and expected number of deploys.")
+      r                         <- validate
```

Comment:
A blank line above and below the extracted method would greatly help with readability. Also, this should be a separate commit saying "Extract `validate` method". I'd then maybe refrain from checking out the PR and just do the review in github :)

Reply:
> should be a separate commit saying "Extract validate method"

Notice I only know that because I did a diff in IDEA :)

Reply:
would you prefer it being a separate top level method?

Reply:
no preference regarding that :)